### PR TITLE
ci: add testing on latest

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -21,12 +21,33 @@ jobs:
...
+
+        - server: mir_test_tools
+          channel: 24/edge
+
+        - server: mir_demo_server
+          channel: null
+
+        - server: mir_demo_server
+          channel: null
+          ppa: mir-team/dev
```
Comment (tarek-y-ismail) [inline .github/workflows/test.yml:38]: Why have `mir_demo_server` listed twice with and without a ppa? I assume the one without a ppa uses the stable version and the one with uses the most recent one?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -21,12 +21,33 @@ jobs:
...
+
+        - server: confined_shell
+          channel: edge
+
+        - server: gnome_shell
+          channel: null
+
         - server: null
+          channel: null
           mark: self
```
Comment (tarek-y-ismail) [inline .github/workflows/test.yml:51]: What does this do?

**Review by tarek-y-ismail — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Saviq) [reply .github/workflows/test.yml:38]: Exactly, yes.

File: .github/workflows/test.yml (hunk 2 above)
Comment (Saviq) [reply .github/workflows/test.yml:51]: This runs only the self tests, those marked with `@pytest.mark.self`. And measures coverage on them. So no server is involved - hence no channel.

**Pull request merged**